Source/EventAnalysis/KTTrackProcessingWeightedSlope.cc

```diff
@@ -0,0 +1,250 @@
+/**
+ @file KTTrackProcessingWeightedSlope.cc
+ @brief Contains KTTrackProcessingWeightedSlope
+ @details Extracts physics-relevant information about tracks using a weighted slope algorithm
+ @author: N.S. Oblath, B. LaRoque & M. Guigue
+ @date: July 22, 2013
+ */
+
+#include "KTTrackProcessingWeightedSlope.hh"
+
+#include "KTHoughData.hh"
+#include "KTLogger.hh"
+
+#include "KTProcessedTrackData.hh"
+#include "KTSmooth.hh"
+#include "KTSparseWaterfallCandidateData.hh"
+#include "KTSequentialLineData.hh"
+
+#include <limits>
+#include <vector>
+#include <algorithm>
+#include <numeric>
+
+using boost::shared_ptr;
+using std::vector;
+using std::string;
+
+namespace Katydid
+{
+    KTLOGGER(tlog, "KTTrackProcessingWeightedSlope");
+
+    // Register the processor
+    KT_REGISTER_PROCESSOR(KTTrackProcessingWeightedSlope, "track-proc-ws");
+
+    KTTrackProcessingWeightedSlope::KTTrackProcessingWeightedSlope(const std::string& name) :
+            KTProcessor(name),
+            fSlopeMinimum(-std::numeric_limits< double >::max()),
+            fProcTrackMinPoints(0),
+            fProcTrackAssignedError(0.),
+            fTrackSignal("track", this),
+            fSWFSlot("swfc", this, &KTTrackProcessingWeightedSlope::ProcessTrack<KTSparseWaterfallCandidateData>, &fTrackSignal),
+            fSeqSlot("seqc", this, &KTTrackProcessingWeightedSlope::ProcessTrack<KTSequentialLineData>, &fTrackSignal)
+    {
+    }
+
+    KTTrackProcessingWeightedSlope::~KTTrackProcessingWeightedSlope()
+    {
+    }
+
+    bool KTTrackProcessingWeightedSlope::Configure(const scarab::param_node* node)
+    {
+        if (node == NULL) return false;
+
+        SetSlopeMinimum(node->get_value("min-slope", GetSlopeMinimum()));
+        SetProcTrackMinPoints(node->get_value("min-points", GetProcTrackMinPoints()));
+        SetProcTrackAssignedError(node->get_value("assigned-error", GetProcTrackAssignedError()));
+
+        return true;
+    }
+
+    template<typename TracklikeCandidate> 
```

Comment:
this works? For some reason I thought template methods needed to be in the header file.

Reply:
I thought that too, but it seems to work there too. My apologies if I impose an useless standard on you. I don't know if there is a standard already in place for these